lldb/unittests/Target/MemoryTest.cpp

```diff
@@ -367,3 +367,97 @@ TEST_F(MemoryDeathTest, TestReadMemoryRangesWithShortBuffer) {
     ASSERT_TRUE(result.empty());
 #endif
 }
+TEST_F(MemoryTest, TestReadPointersFromMemory) {
+  ArchSpec arch("x86_64-apple-macosx-");
+  Platform::SetHostPlatform(PlatformRemoteMacOSX::CreateInstance(true, &arch));
+  DebuggerSP debugger_sp = Debugger::CreateInstance();
+  ASSERT_TRUE(debugger_sp);
+  TargetSP target_sp = CreateTarget(debugger_sp, arch);
+  ASSERT_TRUE(target_sp);
+  ListenerSP listener_sp(Listener::MakeListener("dummy"));
+  ProcessSP process =
+      std::make_shared<DummyReaderProcess>(target_sp, listener_sp);
+  ASSERT_TRUE(process);
+
+  // Read pointers at arbitrary addresses.
+  llvm::SmallVector<addr_t> ptr_locs = {0x0, 0x100, 0x2000, 0x123400};
+  // Because of how DummyReaderProcess works, each byte of a memory read result
+  // is its address modulo 256:
```

Comment:
This comment (and the DummyReaderProcess impl) describe what is done in `TestReadUnsignedIntgersFromMemory`, but in this test we're reading from a series of addresses that have a low byte of 0, so the returned result will be 64-bits of 0x0 in each case.  How does this `expected_result` get returned?

Reply:
>  so the returned result will be 64-bits of 0x0 in each case

I don't think that's true. `DummyReaderProcess` reads byte by byte. So if the pointer's address starts at 0, the pointer will have 8 bytes, the first byte being zero, the last byte being 7

Reply:
Your comment above describes exactly this - reading 4 bytes from addr 0x5 will get you a UInt32 of 0x05050505.  That's what TestReadUnsignedIntegersFromMemory is doing.  But this method is expecting a UInt64 with different values in each byte pos.

Reply:
ah my bad,  i see you're incrementing `addr` in your writer loop.  I thought it would not mutate that.

Reply:
> ah my bad, i see you're incrementing `addr` in your writer loop. I thought it would not mutate that.

`addr` is _the_ loop variable; the function argument is `vm_addr`, which is not mutated. 
I will try to update that code on a separate PR to change the comment as you described and rename some variables.